hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -202,17 +257,17 @@ struct Vgen {
   void emit(const unwind& i);
 
   // instructions
-  void emit(const addl& i) { a->Add(W(i.d), W(i.s1), W(i.s0), SetFlags); }
-  void emit(const addli& i) { a->Add(W(i.d), W(i.s1), i.s0.l(), SetFlags); }
-  void emit(const addq& i) { a->Add(X(i.d), X(i.s1), X(i.s0), SetFlags); }
-  void emit(const addqi& i) { a->Add(X(i.d), X(i.s1), i.s0.q(), SetFlags); }
+  void emit(const addl& i) { a->Add(W(i.d), W(i.s1), W(i.s0), UF(i.fl)); }
```

Comment:
Oh, I guess this sort of demands that the flags bits be in the instruction structs...